apps/web/pages/v2/settings/my-account/general.tsx

```diff
@@ -110,6 +111,8 @@ const GeneralView = ({ localeProp }: GeneralViewProps) => {
     <Form
       form={formMethods}
       handleSubmit={(values) => {
+        setIs24hClockInLocalStorage(values.timeFormat.value === 12 ? "false" : "true");
```

Comment:
This is the main fix for this issue. Making sure that we update the localstorage when saving your preferences in the profile.